File: leetcode_java/7. Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        int of = ((1 << 31) - 1) / 10;
        while (x != 0) {
            if (Math.abs(res) > ((1 << 31) - 1) / 10) return 0;
            res = res * 10 + x % 10;
            x /= 10;
        }
        return res;
    }
}


// class Main {
//     public static void main(String[] args) {
//         // Create a new Solution instance
//         Solution solution = new Solution();
//         // Get the answer
//         int answer = solution.reverse(-2147483412);
//         // Print the answer
//         System.out.print(answer);
//     }
// }

File: leetcode_python/7. Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        y, res = abs(x), 0
        of = 1 << 31 if x > 0 else (1 << 31) - 1
        while y != 0:
            res = res * 10 + y % 10
            if res > of: return 0
            y //= 10
        return res if x > 0 else -res


s = Solution()
print(s.reverse(-12321321312313))

print(1 << 31)